### PR TITLE
feat(verification): Add apple association file

### PR DESCRIPTION
**Pull request by vbudhram**
To open verification links directly in iOS we need to host the `apple-association-file` under our domain.

Ref: https://github.com/mozilla/fxa-bugzilla-mirror/issues/337

### CODE REVIEW

File: server/lib/routes/get-apple-site-association.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+'use strict';
+
+module.exports = function () {
+  const route = {};
+  route.method = 'get';
+  route.path = '/.well-known/apple-site-association';
+
+
+  route.process = function (req, res) {
+
```
Comment (vladikoff) [inline server/lib/routes/get-apple-site-association.js]: nit: remove this new line

File: server/lib/routes/get-apple-site-association.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+module.exports = function () {
+  const route = {};
+  route.method = 'get';
+  route.path = '/.well-known/apple-site-association';
+
+
```
Comment (vladikoff) [inline server/lib/routes/get-apple-site-association.js]: nit: remove this new line, just have a single

File: server/lib/routes/get-apple-site-association.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          },
+          {
+            'appID': '43AQ936H96.org.mozilla.ios.Fennec',
+            'paths': ['*']
+          }
+        ]
+      }
+    };
+
+    res.json(appleAssociationData);
```
Comment (vladikoff) [inline server/lib/routes/get-apple-site-association.js]: a server test for this endpoint would be good 👍

File: server/lib/routes/get-apple-site-association.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    // charset must be set on json responses.
+    res.charset = 'utf-8';
+
+    const appleAssociationData = {
+      'applinks': {
+        'apps': [],
+        'details': [
+          {
+            'appID': '43AQ936H96.org.mozilla.ios.Firefox',
+            'paths': ['*']
```
Comment (rfk) [inline server/lib/routes/get-apple-site-association.js]: Does this mean we can open *any* accounts.firefox.com links in Firefox?  I'm wondering if we only want to force it for some paths, at least to start.

Comment (vbudhram) [inline server/lib/routes/get-apple-site-association.js]: Yea, I think we should force it for `verify_email` and `complete_signin` paths, which probably covers the majority of our emails.

File: server/lib/routes/get-apple-app-site-association.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    // charset must be set on json responses.
+    res.charset = 'utf-8';
+
+    // From Apple developer docs,
+    // https://developer.apple.com/library/content/documentation/General/Conceptual/AppSearch/UniversalLinks.html#//apple_ref/doc/uid/TP40016308-CH12-SW2
+    // to enabled universal link support for a specific domain you need return a json doc
+    // from `/.well-known/apple-app-site-association` path, describing what apps can open which links.
+    // The structure below enables FxiOS Fennec (developer builds), FirefoxBeta (Test Flight builds)
+    // and Firefox (App store builds) to open FxA links.
```
Comment (shane-tomlinson) [inline server/lib/routes/get-apple-app-site-association.js]: Is the order important? If both Firefox and Firefox Beta are installed, will it prefer Firefox to Firefox Beta?

Comment (vbudhram) [inline server/lib/routes/get-apple-app-site-association.js]: Nope, surprisingly the order does not matter.

**Comment by vbudhram** [comment]
@mozilla/fxa-devs This is ready for review when you get a chance!

File: server/lib/routes/get-apple-app-site-association.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    // charset must be set on json responses.
+    res.charset = 'utf-8';
+
+    // From Apple developer docs,
+    // https://developer.apple.com/library/content/documentation/General/Conceptual/AppSearch/UniversalLinks.html#//apple_ref/doc/uid/TP40016308-CH12-SW2
+    // to enabled universal link support for a specific domain you need to return a json doc
+    // from `/.well-known/apple-app-site-association` path, describing what apps can open which links.
+    // The structure below enables FxiOS Fennec (developer builds), FirefoxBeta (Test Flight builds)
+    // and Firefox (App store builds) to open FxA links.
+    const paths = ['/verify_email', '/complete_signin'];
```
Comment (vladikoff) [inline server/lib/routes/get-apple-app-site-association.js]: 👍

**Pull request merged**